base/linalg/bidiag.jl

```diff
@@ -358,17 +358,30 @@ function check_A_mul_B!_sizes(C, A, B)
     end
 end
 
+# function to get the internal vectors for Bidiagonal and [Sym]Tridiagonal
```

Comment:
Maybe add that this is useful because it avoids the allocation when the diagonal exists